content/en/blog/_posts/2023-10-18-gateway-api-time-to-switch-i2gw/index.md

```diff
@@ -0,0 +1,111 @@
+
+The transition from Ingress API to Gateway API in Kubernetes is driven by advantages and advanced functionalities that the Gateway API offers, with its foundation built on four core principles: a role-oriented approach, portability, expressiveness and extensibility.
+
+### A Role-Oriented Approach
+
+Gateway API employs a role-oriented approach that aligns with the conventional roles within organizations involved in configuring Kubernetes service networking. This approach enables infrastructure engineers, cluster operators, and application developers to collectively address different aspects of the Gateway API.
+
+For instance, infrastructure engineers play a pivotal role in deploying GatewayClasses, cluster-scoped resources that act as templates to explicitly define behavior for Gateways derived from them, laying the groundwork for robust service networking.
+
+Subsequently, cluster operators utilize these GatewayClasses to deploy gateways.
+A Gateway in Kubernetes' Gateway-API defines how external traffic can be directed to Services within the cluster, essentially bridging non-Kubernetes sources to Kubernetes-aware destinations.
+It represents a request for a load balancer configuration aligned with a GatewayClass’ specification.
+The Gateway spec may not be exhaustive as some details can be supplied by the GatewayClass controller, ensuring portability.
+Additionally, a Gateway can be linked to multiple Route references to channel specific traffic subsets to designated services.
+
+Lastly, application developers configure route resources (such as HTTPRoutes), to manage configuration (e.g. timeouts, request matching/filter) and Service composition (e.g. path routing to backends)
+Route resources define protocol-specific rules for mapping requests from a Gateway to Kubernetes Services.
+HTTPRoute is for multiplexing HTTP or terminated HTTPS connections.
+It's intended for use in cases where you want to inspect the HTTP stream and use HTTP request data for either routing or modification, for example using HTTP Headers for routing, or modifying them in-flight.
+
+![The resources of Gateway API](gateway-api-resources.svg)
+
+### Portability
+Gateway API is designed to be more portable across different implementations, clusters and environments. It helps reduce Ingress' reliance on non-portable, provider-specific annotations, making your configurations more consistent and easier to manage across multiple clusters.
+
+### Expressiveness
+The Gateway API sets Kuberentes standard support of a wide range of features, such as header-based matching, traffic splitting, weight-based routing, request mirroring and more. With Ingress, these features need custom provider-specific annotations.
+
+### Extensibility
+Gateway API allows for custom resources to be linked at various layers of the API. This makes granular customization possible at the appropriate places within the API structure.
```

Comment:
```suggestion
Gateway API is designed with extensibility as a core feature. Rather than enforcing a one-size-fits-all model, it offers the flexibility to link custom resources at multiple layers within the API's framework. This layered approach to customization ensures that users can tailor configurations to their specific needs without overwhelming the main structure. By doing so, the Gateway API facilitates more granular and context-sensitive adjustments, allowing for a fine-tuned balance between standardization and adaptability. This becomes particularly valuable in complex cloud-native environments where specific use cases require nuanced configurations.
```